base/applications/mspaint/winproc.cpp

```diff
@@ -575,6 +577,42 @@ LRESULT CMainWindow::OnSysColorChange(UINT nMsg, WPARAM wParam, LPARAM lParam, B
     return 0;
 }
 
+typedef ULONG (WINAPI *FN_MAPISendMail)(LHANDLE, ULONG_PTR, lpMapiMessage, FLAGS, ULONG);
+
+ULONG CMainWindow::OpenMailerWithAttachmentW(LPCWSTR pszFileName)
+{
+    if (!PathFileExists(pszFileName))
+        return MAPI_E_ATTACHMENT_NOT_FOUND;
```

Comment:
Notice the behaviour MS' own mspaint has:
If the image hasn't been saved already, save it in the user's own temporary-files folder (but doesn't make mspaint remember it saved the file), then attach that temporarily-saved file to the mail.
(This can be observed from the parameters screenshot I attached.)

Reply:
Fixed. Added `g_szTempFile` global variable.